src/org.xtuml.bp.core/models/org.xtuml.bp.core/ooaofooa/Body/Body/Body.xtuml

```diff
@@ -966,6 +966,476 @@ return false;
+ for each statement in statements
+   foundInstance = statement.findDeclaration(
+                             declarationLineNumber: param.declarationLineNumber,
+                            declarationLineColumn: param.declarationLineColumn);
+   if(foundInstance != ::emptyInstance())
+     // there will only be one case where an instance
+     // was found
+     break;
+   end if;
+ end for;
+ return foundInstance;',
+	1,
+	'',
+	"50f475be-a583-4cdb-8a28-e7bebff13698",
+	0);
+INSERT INTO O_TPARM
+	VALUES ("44de3d9a-78fa-46e9-90c6-da99c4e9753f",
+	"47be481e-6e61-46a4-b087-ee4ac0755e44",
+	'declarationLineNumber',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"00000000-0000-0000-0000-000000000000",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("46048ce5-1e6b-4fbc-82ef-b80fc0f8ce3f",
+	"47be481e-6e61-46a4-b087-ee4ac0755e44",
+	'declarationLineColumn',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"44de3d9a-78fa-46e9-90c6-da99c4e9753f",
+	'');
+INSERT INTO O_TFR
+	VALUES ("93b8e8fa-9b9e-414e-a803-ce4f9cc6e12d",
+	"d1c9af7b-9df6-432a-a25e-96cab24fc6af",
+	'findDeclarationOfVariableOrValueInBlock',
+	'',
+	"bc3722c8-5188-4529-8519-cf9eb9816b17",
+	1,
+	'/*
+ * For all variables in this block look for a matching location
+ */
+select any block from instances of ACT_BLK where (selected.Block_ID == param.blockId);
+select many variables related by block->V_VAR[R823];
+for each variable in variables
+  result = self.findVariableLocationForVariable(declarationLineNumber:
+                             param.declarationLineNumber, declarationLineColumn:
+                          param.declarationLineColumn, var_id: variable.Var_ID, checkLocation: true);
+  if(result != ::emptyInstance())
+    return result;
+  end if;
+end for;
+select many values related by block->V_VAL[R826];
+for each value in values
+  result = self.findDeclarationOfValue(declarationLineNumber:
+                           param.declarationLineNumber, declarationLineColumn:
+                           param.declarationLineColumn, val_id: value.Value_ID);
+  if(result != ::emptyInstance())
+    return result;
+  end if;
+end for;
+return ::emptyInstance();',
+	1,
+	'',
+	"47be481e-6e61-46a4-b087-ee4ac0755e44",
+	0);
+INSERT INTO O_TPARM
+	VALUES ("843648db-93dc-4d99-96f3-b71856f50a78",
+	"93b8e8fa-9b9e-414e-a803-ce4f9cc6e12d",
+	'blockId',
+	"ba5eda7a-def5-0000-0000-000000000005",
+	0,
+	'',
+	"00000000-0000-0000-0000-000000000000",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("907fc361-22bf-479c-aca2-7cf65f0b1d86",
+	"93b8e8fa-9b9e-414e-a803-ce4f9cc6e12d",
+	'declarationLineNumber',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"843648db-93dc-4d99-96f3-b71856f50a78",
+	'');
+INSERT INTO O_TPARM
+	VALUES ("5bcb93a8-d6c6-4f4b-97b8-dc1a6a53e093",
+	"93b8e8fa-9b9e-414e-a803-ce4f9cc6e12d",
+	'declarationLineColumn',
+	"ba5eda7a-def5-0000-0000-000000000002",
+	0,
+	'',
+	"907fc361-22bf-479c-aca2-7cf65f0b1d86",
+	'');
+INSERT INTO O_TFR
```

Comment:
Would this operation make more sense on V_VAR?

Reply:
agreed, moved